capstone/src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -1,11 +1,269 @@
+   * @return HashMap<String, String> saves a key-value pair of all neccessary user details derived.
+   */
+  public HashMap<String, String> checkIfUserIsAlreadyRegistered(
+    DataSource pool,
+    UserService userService,
+    String get_user_row_query,
+    HttpServletResponse response
+  )
+    throws SQLException, IOException {
+    HashMap<String, String> userDetails = new HashMap<>();
+    String email = userService.getCurrentUser().getEmail();
+    Boolean isUserLoggedIn = userService.isUserLoggedIn();
+    try (Connection conn = pool.getConnection()) {
+      try (
+        PreparedStatement queryStatement = conn.prepareStatement(
+          get_user_row_query
+        )
+      ) {
+        queryStatement.setString(1, email);
+        ResultSet result = queryStatement.executeQuery();
+        if (result.next() == false) {
+          response.sendRedirect("/userDetails.html");
+          userDetails.put("status_code", "307");
+          return userDetails;
+        } else {
+          do {
+            String user_status = Boolean.toString(isUserLoggedIn);
+            userDetails.put("user_status", user_status);
+            String id = Integer.toString(result.getInt("id"));
+            userDetails.put("id", id);
+            userDetails.put("email", email);
+            String full_name = result.getString("full_name");
+            userDetails.put("full_name", full_name);
+            String user_name = result.getString("user_name");
+            userDetails.put("user_name", user_name);
+            String university = result.getString("name");
+            userDetails.put("university", university);
+            String logoutUrl = userService.createLogoutURL(
+              URL_TO_REDIRECT_AFTER_USER_LOGS_OUT
+            );
+            userDetails.put("logoutUrl", logoutUrl);
+            userDetails.put("status_code", "200");
+          } while (result.next());
+          return userDetails;
+        }
+      }
+    }
+  }
+
+  /**
+   * This method is called when a GET request is sent to the "/users" endpoint
+   * @param request A HttpServletRequest object that holds the information sent by the user.
+   * @param response A HttpServletResponse object that holds the information that will be sent back to the user.
+   * @return void
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+    throws IOException, ServletException {
+    response.setContentType("application/json");
+    ServletContext servletContext = getServletContext();
+    DataSource pool = (DataSource) servletContext.getAttribute("my-pool");
+
+    UserService userService = UserServiceFactory.getUserService();
+    try {
+      HashMap<String, String> userDetails = getUserDetails(
+        pool,
+        userService,
+        GET_USER_ROW_QUERY,
+        response
+      );
+      String userResult = new Gson().toJson(userDetails);
```

Comment:
move these out of try.

Reply:
I don't really get this, can you explain this more?

Reply:
I don't know but, if I move line 201 out of the try block, I won't be able to access the userDetails created.